src/hammer-vlsi/par/openroad/__init__.py

```diff
@@ -0,0 +1,231 @@
+    # par main steps
+    #========================================================================
+    def init_design(self) -> bool:
+        # TODO: currently hardlinking the syn outputs, otherwise the 
+        # OpenROAD-flow's default makefile will rebuild syn in this directory
+        syn_dirs = {
+          "results": self.syn_results_path(),
+          "objects": self.syn_objects_path()
+        }
+        for syn_dir in ["results", "objects"]:
+          for src in glob.glob(os.path.join(syn_dirs[syn_dir], "*")):
+              dst = "{syn_dir}/{tech}/{name}/{base}".format(
+                  syn_dir=syn_dir,
+                  tech=self.get_setting("vlsi.core.technology"),
+                  name=self.top_module,
+                  base=os.path.basename(src))
+              self.cmds += [
+                  "rm -f {}".format(dst), 
+                  "ln {} {}".format(src, dst)
+              ]
+        return True
+
+    def floorplan_design(self) -> bool:
+        # TODO: currently using OpenROAD's default floorplan script
+        self.cmds += [
+          "make DESIGN_CONFIG={conf} -f {make} floorplan".format(
+            conf=self.design_config_path(), 
+            make=self.openroad_flow_makefile_path()
+        )]
+        return True
+
+    def place_bumps(self) -> bool:
+        # TODO: currently using OpenROAD's default floorplan script
+        return True
+
+    def place_tap_cells(self) -> bool:
+        # TODO: currently using OpenROAD's default floorplan script
+        return True
+
+    def power_straps(self) -> bool:
+        # TODO: currently using OpenROAD's default floorplan script
+        return True
+
+    def place_pins(self) -> bool:
+        # TODO: currently using OpenROAD's default floorplan script
+        return True
+
+    def place_opt_design(self) -> bool:
+        # TODO: currently using OpenROAD's default place script
+        self.cmds += [
+          "make DESIGN_CONFIG={conf} -f {make} place".format(
+            conf=self.design_config_path(), 
+            make=self.openroad_flow_makefile_path()
+        )]
+        return True
+
+    def clock_tree(self) -> bool:
+        # TODO: currently using OpenROAD's default cts script
+        self.cmds += [
+          "make DESIGN_CONFIG={conf} -f {make} cts".format(
+            conf=self.design_config_path(), 
+            make=self.openroad_flow_makefile_path()
+        )]
+        return True
+
+    def add_fillers(self) -> bool:
+        # TODO: currently using OpenROAD's default cts script
+        return True
+
+    def route_design(self) -> bool:
+        # TODO: currently using OpenROAD's default route script
+        self.cmds += [
+          "make DESIGN_CONFIG={conf} -f {make} route".format(
+            conf=self.design_config_path(), 
+            make=self.openroad_flow_makefile_path()
+        )]
+        return True
+
+    def opt_design(self) -> bool:
+        # TODO: currently no analagous OpenROAD default script
```

Comment:
I think we should have this wording for the other a few others like add_fillers and write_regs

Reply:
done